### ISSUE TITLE
Null Pointer Dereference with lambda experssion for NP_ALWAYS_NULL

### ISSUE THREAD

**Issue by mohui1999** [spotbugs/spotbugs#2436]
## Version
4.7.4

## Rule
NP_ALWAYS_NULL: Null pointer dereference

## Description
The code contains instances where a null pointer is being accessed, which can potentially result in a NullPointerException during code execution. The rule NP_ALWAYS_NULL in Spotbugs detects this error when s.equals() is used in an anonymous expression, but it may not be reported in a lambda expression. This suggests a potential false negative, as the transformation from an anonymous expression to a lambda expression might affect the detection of null pointer issues.

## Code

```java
package googleBugs;

public class Issue10 {

    private void testLambdaClass() {
        Thread t = new Thread(() -> {
            String s = null;
            s.equals(new Object() {      // Not report

                void test() {
                    String s = null;
                    s.equals(new Object() {   // report
                    });
                }
            });
        });
    }
}

```
Correctly reported code
```java
package googleBugs;

private void testAnonymousClass() {
        Thread t = new Thread(new Runnable() {
            @Override
            public void run() {
                String s = null;
                s.equals(new Object() {   // report
                    void test() {
                        String s = null;
                        s.equals(new Object() {    // report

                        });
                    }
                });
            }
        });
    }
```

**Comment by gtoison** [comment]
I've tried reproducing the problem here: https://github.com/spotbugs/spotbugs/commit/5a4904c00179b0a889a687a01abc0789949a6949 but the bug seems to be detected.
In the test the paths of the `.class` files are hardcoded, maybe you can check what classes are analysed?
